exchanges/okx/okx_types.go

```diff
@@ -3103,14 +4137,14 @@ type OfferInvestData struct {
 // PurchaseRequestParam represents purchase request param specific product
 type PurchaseRequestParam struct {
 	ProductID  string                   `json:"productId"`
-	Term       int                      `json:"term,string,omitempty"`
+	Term       int64                    `json:"term,string,omitempty"`
 	InvestData []PurchaseInvestDataItem `json:"investData"`
 }
 
 // PurchaseInvestDataItem represents purchase invest data information having the currency and amount information
 type PurchaseInvestDataItem struct {
-	Currency string       `json:"ccy"`
-	Amount   types.Number `json:"amt"`
+	Currency currency.Code `json:"ccy"`
+	Amount   float64       `json:"amt,string"`
```

Comment:
this can be reverted to a types.Number?

Reply:
It is being used as a parameter value... Not as a response. That's why it is of type _float64_ but not _types.Number_